pipeline/stream_parser.go

```diff
@@ -262,11 +262,20 @@ func NewMessageProtoParser() (m *MessageProtoParser) {
 
 func (m *MessageProtoParser) Parse(reader io.Reader) (bytesRead int, record []byte, err error) {
 	if m.needData {
-		if bytesRead, err = m.read(reader); err != nil {
+		bytesRead, err = m.read(reader)
+		m.readPos += bytesRead
+		if err == io.EOF {
```

Comment:
None of this code needs to be duplicated. If you have bytes read and err was not nil and not EOF return otherwise fall through.